.github/workflows/ci.yml

```
@@ -304,6 +304,41 @@ jobs:
             dist-newstyle
           key: ${{ env.CABAL_CACHE_VERSION }}-cabal-${{ matrix.os }}-${{ matrix.ghc }}-${{ hashFiles(format('cabal.GHC-{0}.config', matrix.ghc)) }}-${{ github.sha }}
 
+  # The source archives that github autogenerates for releases do not
+  # include submodules, and also do not include submodule references, so
+  # they are unbuildable and broken. See issue 2532.
+  #
+  # We only need one of these, and it can run anywhere; nothing in it
+  # is target-dependent.
+  source-archive:
+    runs-on: ubuntu-24.04
+    needs: [config]
+    # Run this piece only on tag builds, which in practice means release
+    # tags only. Skip running it on forks as well, because we won't have
+    # the signing key there.
+    if: github.event_name == 'push' && needs.config.outputs.tag == 'true' && github.repository_owner == 'GaloisInc'
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v4
+      - run: |
+           git submodule update --init
```

Not your fault (I see you copy-pasted this from elsewhere), but the `|` part is unnecessary due to there only being a single command in this step. We could just as well do:

```suggestion
      - run: git submodule update --init
```